# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../crypto++/libs-only.patch
# Copyright (C) 2010 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

Description:
  This patch is changing the crypto++ makefile to build both, a static
  and a shared version of the library, but not the supplied test
  application.

TODO:
  There is quite some room for improvements inside of the makefile, so
  it would be good to have an more "inteligent" way to select what kind
  of library should be build (i.e. static and shared, static-only,
  shared-only).
  Besides that the 'install' target is very poorly implemented and should
  be improved to consider the variants mentioned above!

--- ./GNUmakefile.orig	2010-08-09 14:22:42.000000000 +0200
+++ ./GNUmakefile	2010-08-25 12:02:14.476057984 +0200
@@ -131,7 +131,11 @@
 TESTIMPORTOBJS = $(TESTOBJS:.o=.import.o)
 DLLTESTOBJS = dlltest.dllonly.o
 
-all: cryptest.exe
+all: static shared
+
+static: libcryptopp.a
+
+shared: libcryptopp.so
 
 test: cryptest.exe
 	./cryptest.exe v
@@ -144,7 +148,6 @@
 	$(CP) *.h $(PREFIX)/include/cryptopp
 	$(CP) *.a $(PREFIX)/lib
 	$(CP) *.so $(PREFIX)/lib
-	$(CP) *.exe $(PREFIX)/bin
 
 libcryptopp.a: $(LIBOBJS)
 	$(AR) $(ARFLAGS) $@ $(LIBOBJS)
